### ISSUE TITLE
Upgrade JavaParser dependency

### ISSUE THREAD

**Issue by matozoid** [timowest/scalagen#86]
The latest version is much more stable and has support for Java 9.

https://github.com/javaparser/javaparser/

**Comment by hoijui** [comment]
@matozoid could you give some hints for updating from current javaparser 2.5.1 to the latest? it looks like the API changed a lot (i got something like 100+ compile errors for scalagen).
.. or would i just have to dig through the docs until i understand it, if i were to update this?

**Comment by hoijui** [comment]
.. woops! :D
https://github.com/javaparser/javaparser/wiki/Migration-Guide

**Comment by matozoid** [comment]
Cool :-) I'll help out if it gets too complex.

**Comment by hoijui** [comment]
aehm... i tried, but there are over 600 errors, and while i know my way around in java well, i am a complete beginner in scala, and.. fixing one error might easily take me 5 to 30min, and i probably get to fix it in a very ugly way ... it really makes no sense, as someone else would have to go over it anyway, and it would take me at least a month full time to get it compiling again.

**Comment by matozoid** [comment]
Most of these will probably be because of `Optional` - so that should be quickfixable by adding `.get()` there.

**Comment by hoijui** [comment]
i only looked at the first two files with errors, and there nearly everything was about NodeList

**Comment by matozoid** [comment]
What about #84 by the way?

**Comment by hoijui** [comment]
that is actually what i look at/work on. i made some patches on top of that, which nightscape already accepted, here: https://github.com/nightscape/scalagen/tree/functional_visitor
but they are not related to the java parser update.
.. yeah, i consider this branch of him to be latest scalagen development version.

**Comment by matozoid** [comment]
I'm going through the errors. Most are simple name changes (getStmts -> getStatements). Some are annoying errors because we move to a specific list type (ArrayList -> NodeList). A few are because of Optional usage. A bunch are because we no longer have a bit field for modifiers, but an EnumSet.
  
I haven't done anything with Scala for a few years now, so I'll just try to reduce the error count as much as possible.

**Comment by matozoid** [comment]
I'm over here: https://github.com/matozoid/scalagen/tree/jp_3

**Comment by hoijui** [comment]
ohh nice! :-)
yeah.. if it is too much work, and you know you will take a break.. maybe i will have a try, now that you gave good examples in your wipwip commit for how to do many of the changes.
did you also consider writing a simple script, for at least the name-only changes, which could be run over all the sources at once... or did you already do that?

**Comment by matozoid** [comment]
Name changes are the most satisfying to refactor by hand, so easy ;-) But yeah, just a global search/replace in the IDE would help.

I'm having a break now!

**Comment by hoijui** [comment]
hehe :D
awww thats very true! :D

ook, i will let you know if i start working on it, but i am traveling
right now for a few days, so i might not get around to it.
but cool you started!

btw, is scala still allive? just because i see many related projects
(like scalagen) when the main repos have not seen updates since 3+
years.
i am just entering the scene, as scala sounds like the first thing
with which i would want to replace my home base/language of choice
(java), but this lets me question it a bit.

2018-01-07 22:21 GMT+01:00, Danny van Bruggen <notifications@github.com>:

**Comment by hoijui** [comment]
hmm.. ok you said, you are also not into scala anymore..

2018-01-07 22:41 GMT+01:00, hoijui <hoijui.quaero@gmail.com>:

**Comment by matozoid** [comment]
It is very much alive. What I hear from rumours in the consultancy company I work for seems to be that a few percent of software in The Netherlands is built with it. I could easily be off a magnitude though ;-)

**Comment by hoijui** [comment]
oh wow :-)
... oh.. so maybe its just a few per-mill, hm?

i just found this:
http://githut.info/
(language statistics of github repos)
.. indeed scala is quite big! much more then i though.

2018-01-08 19:13 GMT+01:00, Danny van Bruggen <notifications@github.com>:

**Comment by matozoid** [comment]
So will you work on this more then? :-)

**Comment by hoijui** [comment]
hmm...
my live is currently very chaotic. i have no own place to stay, am
currently traveling and staying wiht friends, trying to find
eco-villages (currently in the south of iberian island) to spend some
time (months). some days i am depressed, others i am high. i want to
do physical work, but i soehow fail to get there, and sometimes i want
to flee into programming, and sometimes i even enjoy it but.. every
day is different, and i don't know how my live will look like on
monday. for example.
right now (until monday), i am with a friend on the countryside, and
there is stuf to do outside.. planting stuff, setting up kompost, ...
so i do that whenever possible.
ah well, too muhc detail, but.. i just can't say. :/
i do have your latest changes merged and ideaJ open t owokr on it though. ;-)

2018-01-09 18:54 GMT+01:00, Danny van Bruggen <notifications@github.com>:

**Comment by matozoid** [comment]
Yeah, too much detail, but good luck anyway ;-) Let me know if you have trouble figuring some things out.

**Comment by hoijui** [comment]
:D
i sure will, thanks!
and as said, i will tell you when i start working on it.

2018-01-13 19:54 GMT+01:00, Danny van Bruggen <notifications@github.com>:

**Comment by nightscape** [comment]
Hi @matozoid and @hoijui , I've forked this project to https://github.com/scalagen/scalagen
@matozoid I saw that you already made some changes on top of my fork.
Would you mind opening a PR?
